Tests/SentryTests/Helper/SentrySerializationTests.swift

```diff
@@ -267,6 +267,17 @@ class SentrySerializationTests: XCTestCase {
         XCTAssertNil(actual)
     }
     
+    func testReturnNilForCorruptedEnvelope() throws {
+        let envelope = SentryEnvelope(event: Event(error: NSError(domain: "test", code: -1, userInfo: nil)))
+        let data = try XCTUnwrap(SentrySerialization.data(with: envelope))
+        
+        let corruptedData = data[0..<data.count - 1]
+        
+        let unserialized = SentrySerialization.envelope(with: corruptedData)
```

Comment:
`m`: maybe add a happy path version of the same test where we just deserialize the data object like `SentrySerialization.envelope(with: data)` (didn't find a test that does exactly this)

Reply:
We have other tests that serialize and deserialize the envelope, like `testSentryEnvelopeSerializer_WithSingleEvent`